File: javascript/bubble_sort.js
function bubbleSort(arr){
	
  for(let i = 0; i < arr.length; i++){


    for(let j = 0; j < ( arr.length - i -1 ); j++){

      if(arr[j] > arr[j+1]){
        let temp = arr[j]
        arr[j] = arr[j + 1]
        arr[j+1] = temp
      }
    }
  }
  console.log(arr);
}

var arr = [95, 34, 65, 89, 2, 78, 12, 8, 57];

bubbleSort(arr);

File: C++/Bubble_sort.cpp
// C++ program for implementation of Bubble sort
#include <iostream>
#include <conio.h>
#include <stdlib.h>

using namespace std;

void swap(int *a, int *b)
{
	int temp = *a;
	*a = *b;
	*b = temp;
}

// A function of bubble sort
void bubbleSort(int arr[], int n)
{
	int i, j;
	for (i = 0; i < n-1; i++)

	// Last i elements are already in place
	for (j = 0; j < n-i-1; j++)
		if (arr[j] > arr[j+1])
			swap(&arr[j], &arr[j+1]);
}

/* Function to print an array */
void printArray(int arr[], int size)
{
	int i;
	for (i = 0; i < size; i++)
		cout << arr[i] << " ";
	cout << endl;
}

// main code
int main()
{
	int arr[] = {77, 32, 20, 42, 65, 11, 90};
	int n = sizeof(arr)/sizeof(arr[0]);
	bubbleSort(arr, n);
	cout<<"Sorted array: \n";
	printArray(arr, n);
	return 0;
}

// This code is contributed by mohit
